lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -277,6 +277,7 @@ def add_test_targets
                 # requires frameworks. For tests we always use the test target name as the product name
                 # irrelevant to whether we use frameworks or not.
                 configuration.build_settings['PRODUCT_NAME'] = name
+                configuration.build_settings['PRODUCT_MODULE_NAME'] = name
```

Comment:
Thank you again for the update.

Just tested on a sample pod, I thought the default value would work, but Xcode fails:
with `module name "ImageCoordinateSpace-Unit-Tests" is not a valid identifier`
```
pod lib lint

 -> ImageCoordinateSpace (1.0.5)
    - ERROR | [iOS] xcodebuild: Returned an unsuccessful exit code.
    - NOTE  | [iOS] xcodebuild:  <unknown>:0: error: module name "ImageCoordinateSpace-Unit-Tests" is not a valid identifier
    - NOTE  | [iOS] xcodebuild:  clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

The valid identifier would be `ImageCoordinateSpace_Unit_Tests`
or value like `$(PRODUCT_NAME:c99extidentifier)` which is Xcode default value that always create valid name.
or the even better way would be to clear it, so Xcode will pick the default value.
```
configuration.build_settings.delete('PRODUCT_MODULE_NAME')
```

Reply:
weird, I tried it and it worked for me to build..........will make a follow up.

Reply:
Could be my environment?

Here is the build on travis:
https://travis-ci.org/paulz/ImageCoordinateSpace/jobs/354136986#L2284

and if you want to try locally run these commands:

```
git clone -b test-spec git@github.com:paulz/ImageCoordinateSpace.git
cd ImageCoordinateSpace/
bundle
pod lib lint --verbose
```
and here is what I see:
```
 -> ImageCoordinateSpace (1.0.5)
    - ERROR | [iOS] xcodebuild: Returned an unsuccessful exit code.
    - NOTE  | [iOS] xcodebuild:  <unknown>:0: error: module name "ImageCoordinateSpace-Unit-Tests" is not a valid identifier
    - NOTE  | [iOS] xcodebuild:  clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Reply:
Now works for me on master:

https://travis-ci.org/paulz/ImageCoordinateSpace/builds/356194782#L717
Test pod creates module using Xcode default value:

<img width="816" alt="screen shot showing Xcode default value for module name" src="https://user-images.githubusercontent.com/59230/37695423-60a209a6-2c8c-11e8-9bb5-b4f66beee3de.png">
<img width="1519" alt="screen shot showing produce module name setting across all levels" src="https://user-images.githubusercontent.com/59230/37695424-60b7450a-2c8c-11e8-8592-773320794dc2.png">